src/main/osd/osd_elements.c

```diff
@@ -1082,12 +1087,13 @@ static void osdElementEfficiency(osdElementParms_t *element)
 {
     int efficiency = 0;
     if (sensors(SENSOR_GPS) && ARMING_FLAG(ARMED) && STATE(GPS_FIX) && gpsSol.groundSpeed >= EFFICIENCY_MINIMUM_SPEED_CM_S) {
-        const int speedX100 = osdGetSpeedToSelectedUnit(gpsSol.groundSpeed * 100); // speed * 100 for improved resolution at slow speeds
+        const int speed = osdGetSpeedToSelectedUnit(gpsSol.groundSpeed); 
         
-        if (speedX100 > 0) {
+        if (speed > 0) {
```

Comment:
Do we really need this condition and static variable `efficiencyF`?
If speed is really == 0 then you will see only the last value not the zero. Also calling `pt1FilterApply` with uneven rate is not good for the filter.

Cutoff `#define EFFICIENCY_CUTOFF_HZ 0.212206591f` is also pretty small and weird number.


We can change `mAmperage` and `speed` type to `float` to avoid explicit type conversion.

Reply:
I changed the cutoff to 0.2, low but adequate.  The filter is just trying to smooth the efficiency OSD display so it's easier to read, it really doesn't matter if it misses a few frames when speed == 0.  Besides we need the divide by zero protection.

Reply:
corrected